### DISCUSSION TITLE
Is it possible to override validation for a specific type

### DISCUSSION THREAD

**Discussion by super-falcon** [networknt/json-schema-validator#1067] [Q&A] [answered]
In my example

I want to be able to make an ignorecase comparison for enum values passed in the input file.
Schema File:
`
{
  "$schema": "http://json-schema.org/draft-06/schema#",
  "$ref": "#/definitions/ProductList",
  "definitions": {
    "ProductList": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "version": {
          "type": "number",
        },
        "list": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/ProductRelease"
          }
        }
      },
      "required": [
        "list",
        "version"
      ],
      "title": "ProductList"
    },
    "ProductRelease": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "productName": {
          "type": "string",
        },
        "packageType": {
          "type": "string",
          "enum": [
            "single",
            "collection"
          ],
          "default": "single",
        },
      },
      "required": [
        "productName",
        "packageType"
      ],
      "title": "ProductRelease"
    },
  }
}
`
The input file
`{
  "version": 1,
  "list": [
    {
      "productName": "test",
      "packageType": "COLLECTION"
    }
  ]
}
`
If you see the input file has value of **COLLECTION** for enum packageType, where as in the schema it is defined as lower case. Looking to see if there is a way for this validation to pass.

**Comment by justin-tay** [comment]
The `enum` keyword is case-sensitive. Unless you also want to accept mixed-case variants, I would just add the upper-case enums. eg. 

```json
{
  "enum": ["single","collection", "SINGLE", "COLLECTION"]
}
```

You can of course write a custom `enum` keyword implementation that does case-insensitive comparisons, but this will not interoperate with other implementations and is really not recommended. Eg. https://www.jsonschemavalidator.net/s/zXUHfOlv

**Comment by super-falcon** [comment]
Thx Justin.

The yaml form of the input file is read by SNAKEYAML and it needs the enum to be uppercase and hence the issue.

Is this a good example of how I could override enum keyword implementation https://github.com/networknt/json-schema-validator/blob/master/doc/custom-meta-schema.md

**Reply by justin-tay** [reply, marked as answer]
Yes that's a good example.

However what you are saying seems to imply that only uppercase enum values are acceptable which means you should be changing your schema to only accept such values.

**Comment by super-falcon** [comment]
Hi Justin,

    Thx for the reply. 
My use case got solved by having everything uppercase as suggested by you in the previous comment. I.e. in validation, snake yaml etc.
 {
  "enum": ["SINGLE", "COLLECTION"]
}

Thx for your help. I will close this discussion
